core/types/calls.cc

```diff
@@ -3075,14 +3075,17 @@ class Magic_checkAndAnd : public IntrinsicMethod {
                                         "&.");
 
                                     auto funLoc = core::Loc(args.locs.file, args.locs.fun);
-                                    if (funLoc.exists() && !funLoc.empty() &&
+                                    if (selfTyAndAnd.origins.size() == 2 && funLoc.exists() && !funLoc.empty() &&
                                         funLoc.adjustLen(gs, -1, 1).source(gs) == ".") {
-                                        auto andAndLoc = args.locs.args[2];
+                                        // Our checkAndAnd desugarer only fires if the LHS and RHS have matching Send
+                                        // nodes. In these cases, we know that there are two origins and that the first
+                                        // origin is the call on the LHS.
+                                        auto lhsLoc = selfTyAndAnd.origins[0];
```

Comment:
By which you mean that instead of changing the Loc for a single-element InsSeq in desugar, we should parse the node in infer and see if it’s parent node is a single element wrapped in parens?